api/v1alpha1/tls_types.go

```diff
@@ -133,3 +147,36 @@ type ClientValidationContext struct {
 	// +optional
 	CACertificateRefs []gwapiv1.SecretObjectReference `json:"caCertificateRefs,omitempty"`
 }
+
+// StatelessSessionResumptionSettings defines setting for stateless (session-ticket based) session resumption
+// +kubebuilder:validation:XValidation:rule="(self.enabled == false && self.sessionTicketKeysRef.size() > 0)",message="sessionTicketKeysRef can only be set when enabled is true"
+type StatelessSessionResumptionSettings struct {
+	// Enabled determines if stateless session resumption is supported. When Envoy Proxy is deployed with more than
+	// one replica, session ticket encryption keys are not synchronized between instances, possibly leading to resumption
+	// failures. Users can synchronize session ticket encryption keys be storing them in a secret and using the
+	// sessionTicketKeysRef option.
+	// Note that improper handling of session ticket encryption keys may result in loss of secrecy.
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#extensions-transport-sockets-tls-v3-tlssessionticketkeys
+	// Default: true
+	Enabled *bool `json:"enabled,omitempty"`
```

Comment:
If we add an intermediate layer, `Enabled` can be removed.

```
SessionResumption:
  Type: Stateless
  keys:
  - ...
```

Reply:
Good input, I adopted this format.

Reply:
Rethink on this,  a fixed list of 2 elements seems a bit strange, what about the following structure? cc @arkodg always has great idea on how to shape the API.

 ```yaml
 SessionResumptionSettings:
    stateful: {}
    stateless: {}
```